tests/unit_tests/test_array_apply.py

```diff
@@ -0,0 +1,36 @@
+import json
+
+import pytest
+
+from tests.utils import load_process_code, run_process, run_input_validation
+
+
+@pytest.fixture
+def array_apply_process_code():
+    return load_process_code("array_apply")
+
+
+@pytest.mark.parametrize(
+    "example_input,expected_output", 
+    [
+        ({"data": [], "process": {"process_graph": {"add_pg": {"process_id": "add", "arguments": {"x": {"from_parameter": "x"}, "index": {"from_parameter": "index"}, "y": 10}}}}}, [])
```

Comment:
I don't think this test does anything. When running unit tests the JS code is run directly, it doesn't go through the converter that would generate a JS function from the `process` argument. 
Thus, the `process` argument should be passed as a JS function. Similar to how it's done here: https://github.com/openEOPlatform/openeo-pg-evalscript-converter/pull/57/files

Reply:
This one in particular does not do much yes, it just propagates the empty array. However the `array_apply` process function checks for both parameters `data` and `process` which are required and are array/object. That is why there is an actual process_graph object as the parameter instead of some dummy text or something like that.

I added a few more tests similar to how you proposed.

Reply:
I'd say that since `process` parameter is not used it would be clearer if some dummy text would be passed. Passing a process graph that is not parsed or used might be misleading.

On that note, I am not entirely sure that we've taken the best approach by validating the `process` parameter (the same as in the `array_filter`). We might rather (or possibly also) check if `process()` function is defined and act upon that.

Reply:
> I'd say that since process parameter is not used it would be clearer if some dummy text would be passed. Passing a process graph that is not parsed or used might be misleading.
On that note, I am not entirely sure that we've taken the best approach by validating the process parameter (the same as in the array_filter). We might rather (or possibly also) check if process() function is defined and act upon that.

What we could also do is change `write_process` like this:
```python
def write_process(self):
        ...
        return f"""
function array_apply(arguments) {{
    function process(arguments) {{
        ... 
    }}
    ...
    const { data, context } = arguments;
    return array_apply({ data: data, process: process, context: context })
}}
"""
```
And then we can pass the `process` as a function directly in unit tests. We can then also check if `process` is defined and callable.

Reply:
This would obviously be ideal. Let's go with this.

Reply:
Should be fixed now.